sys/ktest.c

```diff
@@ -0,0 +1,175 @@
+#include <ktest.h>
+#include <stdc.h>
+#include <malloc.h>
+
+/* Borrowed from mips/malta.c */
+char *kenv_get(const char *key);
+
+/* Stores currently running test data. */
+static test_entry_t *current_test = NULL;
+/* A null-terminated array of pointers to the tested test list. */
+static test_entry_t **autorun_tests;
+
+int ktest_test_running_flag = 0;
+
+/* We only need this pool to allocate some memory for performing operations on
+ * the list of all tests. */
+static MALLOC_DEFINE(test_pool, "test data pool");
```

Comment:
I agree the framework should not use malloc if possible - I originally suspected I'll need more memory space than just that list of pointers. However, I would suggest a different solution - a statically allocated space for the array. It doesn't have to be very big, it's just ~1 ptr per each test. The kernel will check if the current size is large enough to store all pointers (we would manually increase it if required). This way the linker set will stay `const`, array management will be much simpler, and we'll be able to e.g. run each test multiple times in a shuffled order (a feature I'll implement in a subsequent PR).